whisper.cpp

```diff
@@ -6055,6 +6055,16 @@ float whisper_full_get_token_p(struct whisper_context * ctx, int i_segment, int
     return ctx->state->result_all[i_segment].tokens[i_token].p;
 }
 
+struct ggml_tensor * whisper_get_encoder_embedding(struct whisper_context * ctx)
+{
+    return ctx->state->embd_enc;
+}
+
+struct ggml_tensor * whisper_get_encoder_embedding_from_state(struct whisper_state * state)
+{
+    return state->embd_enc;
+}
+
```

Comment:
Is it possible to integrate `ggml_backend_tensor_get()` within `whisper_get_encoder_embedding()` so that we can bypass the need to handle `ggml_tensor` directly?

```suggestion
void whisper_get_encoder_embedding(struct whisper_context * ctx, void * data) {
    ggml_tensor * tensor = ctx->state->embd_enc;
    ggml_backend_tensor_get(tensor, data, 0, ggml_nbytes(tensor));
}

void whisper_get_encoder_embedding_from_state(struct whisper_state * state, void * data) {
    ggml_tensor * tensor = state->embd_enc;
    ggml_backend_tensor_get(tensor, data, 0, ggml_nbytes(tensor));
}

```

Reply:
Alright, I see the flaw in my suggestion. It makes the API ineffective because you can't determine the buffer size without first knowing the size of `embd_enc`. If you already know the size, why not add an extra line to fetch the data directly? It seems like we're going around our elbow to get to our thumb : )